src/Compilers/Core/Portable/DiagnosticAnalyzer/Suppression.cs

```diff
@@ -48,5 +50,34 @@ public static Suppression Create(SuppressionDescriptor descriptor, Diagnostic su
         /// Diagnostic suppressed by this suppression.
         /// </summary>
         public Diagnostic SuppressedDiagnostic { get; }
+
+        public static bool operator ==(Suppression left, Suppression right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(Suppression left, Suppression right)
+        {
+            return !(left == right);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is Suppression suppression
+                && Equals(suppression);
+        }
+
+        public bool Equals(Suppression other)
+        {
+            return EqualityComparer<SuppressionDescriptor>.Default.Equals(Descriptor, other.Descriptor)
```

Comment:
>EqualityComparer<SuppressionDescriptor>.Default.Equals(Descriptor, other.Descriptor) [](start = 19, length = 84)

Why go through an extra indirection, it looks like neither Descriptor, nor SuppressedDiagnostic can be null and both expose strongly typed Equals?

Reply:
The containing type is a value type. This pattern ensures the default value for the type does not throw exceptions from `Equals`.